python/src/utils.h

```diff
@@ -44,6 +44,23 @@ inline array to_array_with_accessor(nb::object obj) {
   }
 }
 
+inline bool is_convertable_to_array(const ScalarOrArray& v) {
+  // Checks if the value can be converted to an array (or is already an
+  // mlx array)
+  if (auto pv = std::get_if<nb::bool_>(&v); pv) {
+    return true;
+  } else if (auto pv = std::get_if<nb::int_>(&v); pv) {
+    return true;
+  } else if (auto pv = std::get_if<nb::float_>(&v); pv) {
+    return true;
+  } else if (auto pv = std::get_if<std::complex<float>>(&v); pv) {
+    return true;
+  } else if (auto pv = std::get_if<nb::object>(&v); pv) {
+    return nb::isinstance<array>(*pv) || nb::hasattr(*pv, "__mlx_array__");
+  }
```

Comment:
What I mean is that you replace this entire function with:
```c++
if (auto pv = std::get_if<nb::object>(&v); pv) {
    return nb::isinstance<array>(*pv) || nb::hasattr(*pv, "__mlx_array__");
} else {
    return true;
}
```

The only time it's not an object is when it's a python bool, int or complex.

Reply:
Done ✅